pkg/plugins/manager/signature/manifestverifier/verifier.go

```diff
@@ -111,20 +152,73 @@ func (pmv *ManifestVerifier) GetPublicKey(keyID string) (string, error) {
 	}()
 
 	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
-		return "", err
+		return err
 	}
 
 	if len(data.Items) == 0 {
-		return "", errors.New("missing public key")
+		return errors.New("missing public key")
+	}
+
+	cachedKeys, err := pmv.kv.ListKeys(ctx)
+	if err != nil {
+		return err
 	}
 
+	shouldKeep := make(map[string]bool)
 	for _, key := range data.Items {
-		pmv.publicKeys[key.KeyID] = key
+		err = pmv.kv.Set(ctx, key.KeyID, key.PublicKey)
+		if err != nil {
+			return err
+		}
+		shouldKeep[key.KeyID] = true
+	}
+
+	// Delete keys that are no longer in the API
+	for _, key := range cachedKeys {
+		if !shouldKeep[key] {
+			err = pmv.kv.Del(ctx, key)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// Update the last updated timestamp
+	pmv.kv.GetLastUpdated(ctx)
+
+	return nil
+}
+
+// getPublicKey loads public keys from:
```

Comment:
Nit:
```suggestion
// GetPublicKey loads public keys from:
```

Reply:
It looks like this no longer needs to be exported actually 🤔